homeassistant/components/kodi/media_player.py

```diff
@@ -644,6 +677,31 @@ async def async_play_media(self, media_type, media_id, **kwargs):
             await self._kodi.play_playlist(int(media_id))
         elif media_type_lower == "directory":
             await self._kodi.play_directory(str(media_id))
+        elif media_type_lower in [
+            MEDIA_TYPE_ARTIST,
+            MEDIA_TYPE_ALBUM,
+        ]:
+            await self._kodi.clear_playlist()
+            params = {"playlistid": 0, "item": {f"{media_type}id": int(media_id)}}
+            # pylint: disable=protected-access
+            await self._kodi._server.Playlist.Add(params)
+            await self._kodi.play_playlist(0)
+        elif media_type_lower == MEDIA_TYPE_TRACK:
+            await self._kodi.clear_playlist()
+            params = {"playlistid": 0, "item": {"songid": int(media_id)}}
+            # pylint: disable=protected-access
+            await self._kodi._server.Playlist.Add(params)
+            await self._kodi.play_playlist(0)
+        elif media_type_lower in [
+            MEDIA_TYPE_MOVIE,
+            MEDIA_TYPE_EPISODE,
+            MEDIA_TYPE_SEASON,
+            MEDIA_TYPE_TVSHOW,
+        ]:
+            # pylint: disable=protected-access
+            await self._kodi._play_item(
+                {MAP_KODI_MEDIA_TYPES[media_type_lower]: int(media_id)}
```

Comment:
Aren't seasons and TV Shows like artists or albums? They can also contain multiple items.

Reply:
Yes, so far I decided to add a whole season or even tv show to a playlist. If you think that is reasonable I sure can do that.

Reply:
What I meant is why are we using `Playlist.Add` for albums and artists and albums, but `Player.Open` for seasons and shows. Shouldn't this be consistent?

Reply:
Because `Player.Open` works for individual items and in individual episode or movie does not need to be put in a playlist first. But if you'd rather like to put it in a playlist, no problem with me.

Reply:
Maybe I'm missing something here, but a season or a show is not an individual item. I get the individual episode and movie case, but my question wasn't about them.

Reply:
Oh, I see the confusion. I actually decided ___not___ to add a whole season or even tv show to a playlist for now.

Reply:
Please address this in a follow up if needed.